src/dftbp/common/atomicmass.F90

```diff
@@ -93,4 +93,43 @@ elemental function getAtomicMassNumber(number) result(mass)
   end function getAtomicMassNumber
 
 
+  !> Get the atomic symbol given the approximate atomic mass
+  elemental function getAtomicSymbol(mass) result(symbol)
+
+    !> Approximate atomic mass given in a.m.u.
+    real(dp), intent(in) :: mass
+
+    !> Element symbol
+    character(len=2) :: symbol
+
+    integer, parameter :: offset = iachar('a') - iachar('A')
+    integer :: i
+    real(dp) :: delta, minDelta
+
+    if (mass <= 0.0_dp) then
+      symbol = '??'
+      return
+    end if
+
+    minDelta = 1000.0_dp
+    symbol = '  '
+
+    do i = 1, size(elementSymbol)
+      delta = abs(getAtomicMass(elementSymbol(i)) / amu__au - mass)
+      if (delta < minDelta) then
+        minDelta = delta
+        symbol = elementSymbol(i)
+        ! Uppercase first character
+        symbol(1:1) = achar(iachar(symbol(1:1)) - offset)
+      end if
+    end do
+
+    ! Consistency check
+    if (minDelta > 20.0_dp) then
+      symbol = '??'
+    end if
```

Comment:
Again only a style suggestion: if you wish, one could spare the loop and several intermediate variables, using the `elemental` character of `getAtomicMass()`.
```suggestion
     real(dp) :: deltas(size(elementSymbol))
    integer :: iMin

    symbol = '??'
    if (mass <= 0.0_dp) then
      return
    end if

    deltas(:) = abs(getAtomicMass(elementSymbol) / amu__au - mass)
    iMin = minloc(deltas, dim=1)
    if (deltas(iMin) <= 20.0_dp) then
      symbol = elementSymbol(iMin)
      symbol(1:1) = achar(iachar(symbol(1:1)) - offset)
      ! symbol(1:1) = toupper(symbol(1:1))
    end if
```

Reply:
Thank you, I like this.